Gives Polychromic Shorts a Digitigrade Worn Icon State and Modernizes Polychromic Clothing Code

Michiyamenotehifunana opened the pull request:
:cl: Toriate
add: Polychromic shorts now have a digitigrade state
/:cl:

Fixes #10184 but not really because it isn't really an issue to begin with.

Additionally, polychromic clothing code is no longer modular.

![dreamseeker_mXcVTMWK7O](https://user-images.githubusercontent.com/31995558/71308697-b0bb7600-243a-11ea-8314-bf493650f204.png)

Michiyamenotehifunana committed:
Adds a hacky framework for digitigrade polyclothes

also adds a digitigrade state for polychromic shorts

Ghommie commented on modular_citadel/code/modules/clothing/under/polychromic_clothes.dm:
```diff
@@ -28,6 +28,7 @@
 	tertiary_color = "#808080"
 	can_adjust = FALSE
 	mutantrace_variation = NO_MUTANTRACE_VARIATION // because I'm too lazy to port these to digi-compatible and to prove a point from /tg/ whining - Pooj
+	force_alternate_icon = TRUE	//in the rare case that someone actually decides to make digi-compatible stuff, this should come in handy
```
No! 
Instead, set mutantrace_variation to `MUTANTRACE_VARIATION`, change `worn_overlays()` mutable appearances to use icon_file instead of alternate_worn_icon, and move the digitigrade sprites and overlays to `uniform_digi.dmi`! without the damn `_d` suffix!

Ghommie commented on modular_citadel/code/modules/clothing/under/polychromic_clothes.dm:
```diff
@@ -47,6 +48,18 @@
+				H.update_inv_wear_suit()
+			else
+				item_color = initial(item_color)
+
```
FUCK! NO!
Remove all of this!

Ghommie commented on modular_citadel/code/modules/clothing/under/polychromic_clothes.dm:
```diff
@@ -89,6 +102,7 @@
 	secondary_color = "#808080"
 	tertiary_color = "#808080"
 	body_parts_covered = CHEST|GROIN|ARMS
+	mutantrace_variation = MUTANTRACE_VARIATION	//to enable digitigrade wearing
```
Remove this line, once you have followed my other indications.

Ghommie reviewed (changes requested):
This PR actively works against every improvement I have done (or I'm working on) to "mutant race" variations.
This is not the 2018 anymore, and please comply with the requested changes, thank you.

Michiyamenotehifunana committed:
Modernizes Polychromic Uniform Code

No more modular_citadel, but the neck polychrome crap needs more work in another PR

Michiyamenotehifunana committed:
Merge remote-tracking branch 'upstream/master' into polyshorts-for-digis

Michiyamenotehifunana committed:
oh god please don't mess up

Michiyamenotehifunana committed:
git y u do dis

;-;
